### PR TITLE
feat(deps): add timeout setting for dependencies

### PR DESCRIPTION
**Pull request by Tieske**
Also modifies and replaces the SERVICE_DISABLE_HEALTHCHECK option which can now be set by specifying HEALTH_TIMEOUT=0

replaces #551

### CODE REVIEW

File: assets/help/up.txt [hunk 1]
```diff
@@ -32,8 +32,8 @@ Environment variables:
   REDIS_IMAGE      the Redis dependency to use (default redis:6.2.6-alpine)
   SQUID_IMAGE      the Squid dependency to use (default sameersbn/squid:3.5.27-2)
   GRPCBIN_IMAGE    the Grpcbin dependency to use (default moul/grpcbin:latest)
-  SERVICE_DISABLE_HEALTHCHECK
-                   set to 'true' to disable dependency health checks globally
+  HEALTH_TIMEOUT   time in seconds to wait for dependencies to become healthy
```
Comment (StarlightIbuki) [inline assets/help/up.txt:35]: Maybe -1 for `disable`, and 0 for `+inf`?

**Review by StarlightIbuki — APPROVED** [review]

File: pongo.sh [hunk 2]
```diff
@@ -620,43 +635,55 @@ function healthy {
...
+  fi
 
-  msg "Waiting for $dep"
+  msg "Waiting for '$dep' to become healthy"
 
-  while ! healthy "$iid" "$dep"; do
+  local timeout_count=$((HEALTH_TIMEOUT*2))
+  while [ $timeout_count -ge 0 ]; do
     sleep 0.5
+    if healthy "$iid" "$dep"; then
```
Comment (StarlightIbuki) [inline pongo.sh:680]: I'm not sure if it's smarter to return earlier if the docker report's unhealthy (it's waiting for a timeout either way). But this should be good enough.

**Review by StarlightIbuki — APPROVED** [review]

File: assets/help/up.txt (hunk 1 above)
Comment (Tieske) [reply assets/help/up.txt:35]: too complex imho, nobody should need `+inf` and if they do then `365*24*60*60` will probably do.

**Comment by Tieske** [comment]
@StarlightIbuki 🙏 thx!

**Pull request merged**